poetry/core/factory.py

```diff
@@ -419,6 +427,15 @@ def validate(cls, config: dict, strict: bool = False) -> Dict[str, List[str]]:
                                 )
                             )
 
+            # Checking readme file types (must match)
+            if "readme" in config and isinstance(config["readme"], (list, tuple)):
+                readme_types = {content_type(readme) for readme in config["readme"]}
+                if len(readme_types) > 1:
+                    result["errors"] += [
+                        "Declared README files must be of same type: found %s"
```

Comment:
We don't use `%`-formatting anymore. Use an f-string here, and use string.join to comma-separate your list of types.